### PR TITLE
closes #1047 higher order vignette

### LINKED ISSUE #1047: Documentation:  Create a vignette for call_derivation(), restrict_derivation() and slice_derivation()
### Please select a category the issue is focused on?

Function Documentation

### Let us know where something needs a refresh or put your idea here!

Create a vignette to showcase the use of the functions mentioned above. See also discussion in #701.

### PR DESCRIPTION
**Pull request by rossfarrugia**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by rossfarrugia** [comment]
@thomas-neitmann the URL error here is i think expected in NEWS file until this merges to `main` - any advice? i don't want to merge this and have everyone else's PRs fail the URL check because of this

**Comment by bms63** [comment]
We could put that in the .lycheeignore file and try and remember to remove later on?!?  Could go in the Post-release checklist to remove urls that are now valid from the .lycheeignore file.

File: vignettes/higher_order.Rmd
Comment (bundfussr) [inline vignettes/higher_order.Rmd:32]: The list does not render correctly in html.

**Review by bundfussr — CHANGES_REQUESTED** [review]

File: vignettes/higher_order.Rmd
Comment (thomas-neitmann) [inline vignettes/higher_order.Rmd:28]: I'd say only "arguments" rather than "parameters/arguments.

File: vignettes/higher_order.Rmd
Comment (thomas-neitmann) [inline vignettes/higher_order.Rmd:51]: I'd shorten this to just

```r
ae <- convert_blanks_to_na(admiral_ae)
vs <- convert_blanks_to_na(admiral_vs)
```

File: vignettes/higher_order.Rmd
Comment (thomas-neitmann) [reply vignettes/higher_order.Rmd:120]: I'd very much like to see that. The example above is great to introduce the concept as it's just two derivations in one go. However, the true "Power" of `call_derivation()` is much more obvious if you do more than just two derivations at a time.

File: vignettes/higher_order.Rmd
Comment (thomas-neitmann) [inline vignettes/higher_order.Rmd:67]: I think bolding "without" is enough, no need for all uppercse. Same below for "with".

File: vignettes/higher_order.Rmd
Comment (thomas-neitmann) [inline vignettes/higher_order.Rmd:134]: ~~BUT~~ but

File: vignettes/higher_order.Rmd
Comment (thomas-neitmann) [inline vignettes/higher_order.Rmd:174]: ~~BUT~~ but

File: vignettes/higher_order.Rmd
Comment (thomas-neitmann) [inline vignettes/higher_order.Rmd:178]: What I'm missing here is an explanation of `TRUE` being passed to `filter`. ALso an additional examle with three different slices might be good to showcase that the order of the slices matters (if I recall correctly, please correct @bundfussr).

File: vignettes/higher_order.Rmd
Comment (thomas-neitmann) [inline vignettes/higher_order.Rmd:95]: As this is the first of the higher order function I think we should explain the general structure, i.e. the parameters, of the function. Something along the lines of: "`derivation` is the name of the derivation function to apply to the input dataset. Note that no trailing set of parenthesis is required, e.g. `derive_var_extreme_flag` rather than `derive_var_extreme_flag()`. `variable_params` is a `list` of ...

File: vignettes/higher_order.Rmd
Comment (thomas-neitmann) [reply vignettes/higher_order.Rmd:32]: You'll have to insert a line break before the first list item in order to render properly.

**Review by thomas-neitmann — CHANGES_REQUESTED** [review]

File: vignettes/higher_order.Rmd
Comment (bundfussr) [reply vignettes/higher_order.Rmd:178]: @thomas-neitmann , you are correct. The order matters if the conditions are not mutually exclusive (see "Details" section of `slice_derivation()`).

File: vignettes/higher_order.Rmd
Comment (rossfarrugia) [reply vignettes/higher_order.Rmd:120]: sure, can add

File: vignettes/higher_order.Rmd
Comment (rossfarrugia) [reply vignettes/higher_order.Rmd:67]: ok

File: vignettes/higher_order.Rmd
Comment (rossfarrugia) [reply vignettes/higher_order.Rmd:134]: ok

File: vignettes/higher_order.Rmd
Comment (rossfarrugia) [reply vignettes/higher_order.Rmd:174]: ok

File: vignettes/higher_order.Rmd
Comment (rossfarrugia) [reply vignettes/higher_order.Rmd:95]: yep makes sense

File: vignettes/higher_order.Rmd
Comment (koegerr) [reply vignettes/higher_order.Rmd]: Great descriptions Ross. For the slice function, I assume the separately called derivation is the same with different arguments. Or could it also be different derivations per slice?

File: vignettes/higher_order.Rmd
Comment (koegerr) [inline vignettes/higher_order.Rmd:31]: can one only use one function with different arguments per slice or can one call different functions per slice? Would be great to point this out.

File: vignettes/higher_order.Rmd
Comment (koegerr) [inline vignettes/higher_order.Rmd:280]: Ok I think the example is clear, only one function with different arguments. But still might be good to mention

Comment (rossfarrugia) [reply vignettes/higher_order.Rmd:280]: sure, i added the word "single" in select places to make this clearer - let me know if you prefer any more wording to be added

**Review by bundfussr — APPROVED** [review]

**Review by koegerr — APPROVED** [review]

**Comment by thomas-neitmann** [comment]
@oreillb1 Looks like the {admiral} team is good with this new vignette. However, I'd still like to get input from your user perspective. Please take a look and leave your comments. Thank you!

**Comment by rossfarrugia** [comment]
@thomas-neitmann can you approve this one if you're happy with the changes i made and then i'll merge to close this one out? we can always get further feedback anytime and continually improve it but i'm confident it should hit the mark now.

**Review by thomas-neitmann — APPROVED** [review]

**Pull request merged**
